spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/http/GsonHttpMessageConvertersConfiguration.java

```diff
@@ -41,7 +43,7 @@
 
 	@Configuration
 	@ConditionalOnBean(Gson.class)
-	@Conditional(PreferGsonOrMissingJacksonCondition.class)
+	@Conditional(PreferGsonOrMissingJacksonAndJsonbCondition.class)
```

Comment:
What is the rationale behind this. `Gson` is to be used if both Jackson and JSonB are missing? With three choices, the combination are becoming increasingly complex so I wonder if we shouldn't revisit this differently.